go/cmd/dolt/commands/sqlserver/server.go

```diff
@@ -309,3 +305,32 @@ func getConfigFromServerConfig(serverConfig ServerConfig) (server.Config, error,
 
 	return serverConf, nil, nil
 }
+
+// handleProtocolAndAddress returns new server.Config object with only Protocol and Address defined.
+func handleProtocolAndAddress(serverConfig ServerConfig) (server.Config, error) {
+	host := serverConfig.Host()
+	port := serverConfig.Port()
+	serverConf := server.Config{}
+	serverConf.Protocol = "tcp"
+	portAsString := strconv.Itoa(port)
+	hostPort := net.JoinHostPort(host, portAsString)
+	serverConf.Address = hostPort
+	if portInUse(hostPort) {
+		portInUseError := fmt.Errorf("Port %s already in use.", portAsString)
+		return server.Config{}, portInUseError
+	}
+	// if socket is defined with or without value -> unix
+	if serverConfig.Socket() != "" {
+		// unix
```

Comment:
(minor) I'm not sure this comment is adding anything. You could mention that this is a unix socket on line 322, but I dont' think this comment on this line is needed.